test/validation/api/ipsec/ipsec.c

```diff
@@ -1035,16 +1035,21 @@ int ipsec_init(odp_instance_t *inst)
 		fprintf(stderr, "Packet pool creation failed.\n");
 		return -1;
 	}
-	out_queue = odp_queue_create("ipsec-out", NULL);
-	if (ODP_QUEUE_INVALID == out_queue) {
-		fprintf(stderr, "IPsec outq creation failed.\n");
-		return -1;
+	if (mode == ODP_IPSEC_OP_MODE_ASYNC ||
+	    mode == ODP_IPSEC_OP_MODE_INLINE) {
+		out_queue = odp_queue_create("ipsec-out", NULL);
+		if (ODP_QUEUE_INVALID == out_queue) {
+			fprintf(stderr, "IPsec outq creation failed.\n");
+			return -1;
+		}
 	}
 
-	pktio = pktio_create(pool);
-	if (ODP_PKTIO_INVALID == pktio) {
-		fprintf(stderr, "IPsec pktio creation failed.\n");
-		return -1;
+	if (mode == ODP_IPSEC_OP_MODE_INLINE) {
+		pktio = pktio_create(pool);
+		if (ODP_PKTIO_INVALID == pktio) {
+			fprintf(stderr, "IPsec pktio creation failed.\n");
+			return -1;
+		}
```

Comment:
ipsec_term() function seems to need modification as well. Otherwise, it prints "pktio not found" errors for sync/async test cases.


`	pktio = odp_pktio_lookup("loop");
	if (ODP_PKTIO_INVALID != pktio) {
		if (odp_pktio_close(pktio))
			fprintf(stderr, "IPsec pktio close failed.\n");
	} else {
		fprintf(stderr, "IPsec pktio not found.\n");
	}`

Reply:
> 
> 
> ipsec_term() function seems to need modification as well. Otherwise, it prints "pktio not found" errors for sync/async test cases.
> 
> ```
> pktio = odp_pktio_lookup("loop");
> if (ODP_PKTIO_INVALID != pktio) {
> 	if (odp_pktio_close(pktio))
> 		fprintf(stderr, "IPsec pktio close failed.\n");
> } else {
> 	fprintf(stderr, "IPsec pktio not found.\n");
> }
> ```

Will fix this in the next revision.

Reply:
Updated V2 with fix in ipsec_term. With the new change, there is no lookup performed for each resources. Instead, the handles available in the suite_context is used while destroying the resources.

Reply:
@psavol @JannePeltonen Does this change look okay now?